apps/els_lsp/src/els_referl_unsecure_calls_diagnostics.erl

```diff
@@ -0,0 +1,137 @@
+%%==============================================================================
+%% Unsecure Calls by Referl
+%%==============================================================================
+-module(els_referl_unsecure_calls_diagnostics).
+
+%%==============================================================================
+%% Behaviours
+%%==============================================================================
+-behaviour(els_diagnostics).
+
+%%==============================================================================
+%% Exports
+%%==============================================================================
+-export([ is_default/0
+        , run/1
+        , source/0
+        ]).
+
+%%==============================================================================
+%% Includes & Defines
+%%==============================================================================
+-include("els_lsp.hrl").
+-define(MAX_RECURSION_DEPTH, 10).
+-define(TIME_OUT, 3000).
+
+%%==============================================================================
+%% Callback Functions
+%%==============================================================================
+
+-spec is_default() -> boolean().
+is_default() ->
+  true.
+
+-spec run(uri()) -> [els_diagnostics:diagnostic()].
+run(Uri)->
+  run(Uri, 0).
+
+
+-spec run(uri(), number()) -> [els_diagnostics:diagnostic()]. 
+run(Uri, RecursionDepth) when RecursionDepth < ?MAX_RECURSION_DEPTH ->
+  case filename:extension(Uri) of
+    <<".erl">> -> 
+      case referl_node() of
+        disabled ->
+          [];
+        Node -> 
+          case rpc:call(Node, ri, add, [binary_to_list(els_uri:path(Uri))], ?TIME_OUT) of
+            {badrpc, _} ->
+              els_server:send_notification(<<"window/showMessage">>, #{ type => ?MESSAGE_TYPE_ERROR, message => <<"Refactor Erl node is down {badrpc}!">> }),
+              [];
+            error ->
```

Comment:
What's the scenario where a retry can actually help? Also, not sure if we want to really pop a message every second. Maybe use logs instead?